src/components/diff/PlainTextDiff.ts

```diff
@@ -116,40 +155,60 @@ export class PlainTextDiff extends Widget implements Git.Diff.IDiffWidget {
   /**
    * Create wrapper node
    */
-  protected static createNode(
-    baseLabel: string,
-    remoteLabel: string
-  ): HTMLElement {
+  protected static createNode(...labels: string[]): HTMLElement {
     const head = document.createElement('div');
     head.className = 'jp-git-diff-root';
     head.innerHTML = `
     <div class="jp-git-diff-banner">
-      <span>${baseLabel}</span>
-      <span class="jp-spacer"></span>
-      <span>${remoteLabel}</span>
+      ${labels
+        .filter(label => !!label)
+        .map(label => `<span>${label}</span>`)
+        .join('<span class="jp-spacer"></span>')}
     </div>
     <div class="jp-git-PlainText-diff"></div>`;
     return head;
   }
 
   /**
    * Create the Plain Text Diff view
+   *
+   * Note: baseContent will only be passed when displaying
+   *       a three-way merge conflict.
    */
   protected async createDiffView(
     challengerContent: string,
-    referenceContent: string
+    referenceContent: string,
+    baseContent?: string
   ): Promise<void> {
     if (!this._mergeView) {
       const mode =
         Mode.findByFileName(this._model.filename) ||
         Mode.findBest(this._model.filename);
 
-      this._mergeView = mergeView(this._container, {
+      let options: LocalMergeView.IMergeViewEditorConfiguration = {
         value: challengerContent,
         orig: referenceContent,
         mode: mode.mime,
         ...this.getDefaultOptions()
-      }) as MergeView.MergeViewEditor;
+      };
+
+      // Show three-way diff on merge conflict
+      // Note: Empty base content ("") can be an edge case.
+      if (baseContent !== undefined) {
```

Comment:
```suggestion
      if (baseContent != null) {
```

Reply:
```suggestion
      if (baseContent !== null && baseContent !== undefined) {
```

ESLint complaining about `!=` :angry: